mule-extensions-api/src/main/java/org/mule/runtime/extension/api/manifest/ImmutableDescriberManifest.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.manifest;
+
+import static java.util.Collections.unmodifiableMap;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+import java.util.Map;
+
+/**
+ * Immutable implementation of {@link DescriberManifest}
+ *
+ * @since 1.0
+ */
+final class ImmutableDescriberManifest implements DescriberManifest
```

Comment:
This was on extension-support. Moving this class here was actually his idea, alongside with separating the model interfaces in two... anyway, I not only disagree with the concern you propose but I also think it's out of the scope of this PR. I propose discussing this offline and take any corrective action on a separate commit.